corehq/apps/importer/util.py

```diff
@@ -95,6 +97,25 @@ def from_request(cls, request):
 
         )
 
+
+def open_workbook(filename=None, **kwargs):
+    _soft_assert = soft_assert(notify_admins=True)
+    try:
+        return xlrd.open_workbook(filename=filename, **kwargs)
+    except xlrd.XLRDError as e:
+        message = unicode(e)
+        if message == u'Workbook is encrypted':
+            raise ImporterExcelFileEncrypted(message)
+        else:
+            _soft_assert(False, 'displaying XLRDError directly to user', e)
+            raise ImporterExcelError(message)
```

Comment:
Incidentally this does work, although not sure I would recommend it since it could make for some confusing debugging:
```py
import sys
class NewException(Exception): pass
try:
    raise Exception("blow up")
except:
    typ, exc, tb = sys.exc_info()
    try:
        raise Exception("should not be re-raised")
    except:
        pass
    raise NewException, exc, tb
```
output
```
Traceback (most recent call last):
  File "<string>", line 4, in <module>
__main__.NewException: blow up
```
I think it would be clearer to include the exception details in the soft assert. Maybe add an `exc_info` parameter that could be used like so:
```py
_soft_assert(False, 'message', exc_info=sys.exc_info())
```
To follow the convention of the logging package, it may make sense accept `exc_info=True`. Something like:
```py
def _soft_assert(condition, message, exception=None, exc_info=None):
    if exc_info:
        if not isinstance(exc_info, tuple):
            exc_info = sys.exc_info()
        trace = traceback.format_exc()
        message = "{}\n{}".format(message, trace)
    # do other soft assert stuff
```